crates/server/src/store/metrics.rs

```diff
@@ -0,0 +1,69 @@
+use metrics::{Counter, Gauge, Unit, counter, describe_counter, describe_gauge, gauge};
+
+pub(super) struct PersistentStoreMetrics {
+    pub(super) zone_records_total: Gauge,
+    #[cfg(feature = "__dnssec")]
+    pub(super) zone_records_dynamically_updated: Counter,
+}
+
+pub(super) struct QueryStoreMetrics {
+    pub(super) zone_record_lookups: Counter,
+}
+
+pub(super) struct StoreMetrics {
+    pub(crate) query: QueryStoreMetrics,
+    pub(crate) persistent: PersistentStoreMetrics,
+}
+
+impl StoreMetrics {
+    pub(super) fn new(store: &'static str) -> Self {
+        Self {
+            query: QueryStoreMetrics::new(store),
+            persistent: PersistentStoreMetrics::new(store),
+        }
+    }
+}
+
+impl PersistentStoreMetrics {
+    pub(crate) fn new(store: &'static str) -> Self {
+        let zone_records_total = gauge!("hickory_zone_records_total", "store" => store);
+        describe_gauge!(
+            "hickory_zone_records_total",
+            Unit::Count,
+            "number of dns zone records in persisted storages"
+        );
+
+        #[cfg(feature = "__dnssec")]
+        let zone_records_dynamically_updated = {
+            let zone_records_dynamically_updated =
+                counter!("hickory_zone_records_dynamically_updated", "store" => store);
+            describe_counter!(
+                "hickory_zone_records_dynamically_updated",
```

Comment:
The [naming conventions](https://prometheus.io/docs/practices/naming/) suggest this suffix for unit-less accumulating counts.
```suggestion
                counter!("hickory_zone_records_dynamically_updated_total", "store" => store);
            describe_counter!(
                "hickory_zone_records_dynamically_updated_total",
```

Reply:
Fixed.